crates/cext/src/circuit.rs

```diff
@@ -1151,3 +1152,98 @@ pub unsafe extern "C" fn qk_circuit_delay(
 
     ExitCode::Success
 }
+
+/// @ingroup QkCircuit
+/// Convert a given circuit to a DAG.
+///
+/// @param circuit A pointer to the circuit from which to create the DAG.
+/// @param copy_operations Performs a deep copy of operations if set to true
+/// @param qubit_order The order of the qubits in the output DAG. Defaults to the same order
+///     as the input circuit if NULL. If not NULL, it must be of length ``qk_circuit_num_qubits()``.
+/// @param clbit_order The order of the clbits in the output DAG. Defaults to the same order
+///     as the input circuit if NULL. If not NULL, it must be of length ``qk_circuit_num_clbits()``.
+///
+/// @return A pointer to the new DAG.
```

Comment:
Can you add a comment somewhere about the user needing to free the DAG returned by this function later with `qk_dag_free`?

It'd also be good to update the example below.